include/dlaf/eigensolver/band_to_tridiag/mc.h

```diff
@@ -1034,6 +1035,17 @@ TridiagResult<T, Device::CPU> BandToTridiag<Backend::MC, D, T>::call_L(
   // Should be dispatched to local implementation if (1x1) grid.
   DLAF_ASSERT(grid.size() != comm::Size2D(1, 1), grid);
 
+#ifdef DLAF_WITH_HDF5
+  static size_t num_b2t_calls = 0;
```

Comment:
Possibly:
```suggestion
  static std::atomic<std::size_t> num_b2t_calls = 0;
```
and possibly move this into a function-local static in a separate function defined in a `cpp` file so there's only one static. Alternatively to either or both of the above simply document the restrictions somewhere (no concurrent calls, or overlapping indices for different types).

Reply:
As I was scrolling further I saw the warning with the option:
```
WARNING: just a single execution can be dumped on disk, and any subsequent call fails
```
I think this already implies that there can't be concurrent (or even multiple sequential) calls to the algorithm so not sure how important the `atomic` and `static` comment above from me is...

Reply:
The warning is a leftover from the initial implementation by @albestro. I added the `static` variable to allow for multiple (sequential) calls (by writing to a different file).

Reply:
Oh right, you said as much. In that case I suppose the warning should be removed and then I stand by the above comment, at least the part about moving the counter into a separate function so that it's type-independent (I think the `atomic` is less important right now, it's unlikely to be called concurrently).